airflow/serialization/serialized_objects.py

```diff
@@ -586,9 +604,24 @@ def serialize_operator(cls, op: BaseOperator) -> Dict[str, Any]:
         return serialize_op
 
     @classmethod
-    def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> BaseOperator:
+    def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> Union[BaseOperator, MappedOperator]:
         """Deserializes an operator from a JSON object."""
-        op = SerializedBaseOperator(task_id=encoded_op['task_id'])
+        op: Union[BaseOperator, MappedOperator]
+        # Check if it's a mapped operator
+        if "mapped_kwargs" in encoded_op:
```

Comment:
Probably the following might be a better condition ?
```suggestion
        if encoded_op.get("_is_mapped"):
```

Reply:
We don't store _is_mapped currently.

Reply:
The following line does, no?

https://github.com/apache/airflow/blob/3ef6268e0039f40a6380394516bdf4b587bf7a4d/airflow/serialization/serialized_objects.py#L550

Reply:
🤦🏻